### PR TITLE
Add accounts_no_uid_except_zero to RHEL7 STIG

### PR DESCRIPTION
**Pull request by shawndwells**
- add to profile
- update rationale and policy refs

image

### CODE REVIEW

**Comment by redhatrises** [comment]
@shawndwells looks like this build is failing with:
```
File 'output/xccdf-unlinked-unresolved.xml' line 872: Element '{http://checklists.nist.gov/xccdf/1.1}refine-value': Duplicate key-sequence ['var_password_pam_minclass'] in unique identity-constraint '{http://checklists.nist.gov/xccdf/1.1}refineValueKey'.
File 'output/xccdf-unlinked-unresolved.xml' line 880: Element '{http://checklists.nist.gov/xccdf/1.1}select': Duplicate key-sequence ['accounts_password_pam_unix_remember'] in unique identity-constraint '{http://checklists.nist.gov/xccdf/1.1}itemSelectKey'.
OpenSCAP Error: Invalid XCCDF Checklist (1.1) content in output/xccdf-unlinked-unresolved.xml. [oscap_source.c:268]
```

File: RHEL/7/input/profiles/stig-rhel7-server-upstream.xml [hunk 1]
```diff
@@ -30,6 +30,16 @@
      Protection Profile (NIAP OSPP). -->
 
 <!-- Password Requirements -->
+<refine-value idref="var_password_pam_minclass" selector="4" />
```
Comment (isimluk) [inline RHEL/7/input/profiles/stig-rhel7-server-upstream.xml]: This is the cause of the test failure. Please just remove this line.

File: RHEL/7/input/profiles/stig-rhel7-server-upstream.xml [hunk 2]
```diff
@@ -30,6 +30,16 @@
...
+
+<select idref="accounts_maximum_age_login_defs" selected="true" />
+<select idref="accounts_password_pam_unix_remember" selected="true" />
```
Comment (isimluk) [inline RHEL/7/input/profiles/stig-rhel7-server-upstream.xml]: This is the cause of the other test failure. Please just remove this line.

**Comment by shawndwells** [comment]
Thanks @isimluk. Rebase against master, removed merge conflicts and build errors.

`````
[shawnw@devbox-rhel7 7]$ make clean ; make ; make validate
......
oscap xccdf validate-xml output/ssg-sl7-xccdf.xml
oscap ds sds-validate output/ssg-sl7-ds.xml
cd output; ../../../shared/utils/verify-references.py --rules-with-invalid-checks --ovaldefs-unused ssg-rhel7-xccdf.xml
`````

**Comment by mpreisler** [comment]
ACK

**Pull request merged**